src/prognostic_equations/edmfx_sgs_flux.jl

```diff
@@ -194,15 +194,42 @@ function edmfx_sgs_flux_tendency!(
             )
         end
 
+        # mass flux
+        ᶠinterp_u₃ = Operators.InterpolateC2F(
+            top = Operators.SetValue(C3(FT(0))),
+            bottom = Operators.SetValue(C3(FT(0))),
+        )
+        ᶠu³_diff_colidx = p.ᶠtemp_CT3[colidx]
+        for j in 1:n
+            @. ᶠu³_diff_colidx = ᶠu³ʲs.:($$j)[colidx] - ᶠu³[colidx]
+            @. Yₜ.f.u₃[colidx] -= ᶠinterp_u₃(
+                C3(
+                    ᶜρaʲs.:($$j)[colidx] * ᶜdivᵥ(
+                        Geometry.outer(ᶠu³_diff_colidx, ᶠu³_diff_colidx),
+                    ) / Y.c.ρ[colidx],
+                ),
+            )
+        end
         # diffusive flux
         ᶠρaK_u = p.ᶠtemp_scalar
         @. ᶠρaK_u[colidx] = ᶠinterp(Y.c.ρ[colidx]) * ᶠinterp(ᶜK_u[colidx])
+        ᶠstrain_rate = p.ᶠtemp_UVWxUVW
+        compute_strain_rate_face!(ᶠstrain_rate[colidx], ᶜu[colidx])
+        # set boundary momentum fluxes
         ᶜdivᵥ_uₕ = Operators.DivergenceF2C(
             top = Operators.SetValue(C3(FT(0)) ⊗ C12(FT(0), FT(0))),
             bottom = Operators.SetValue(sfc_conditions.ρ_flux_uₕ[colidx]),
         )
-        @. Yₜ.c.uₕ[colidx] -=
-            ᶜdivᵥ_uₕ(-(ᶠρaK_u[colidx] * ᶠgradᵥ(Y.c.uₕ[colidx]))) / Y.c.ρ[colidx]
+        @. Yₜ.c.uₕ[colidx] -= C12(
+            ᶜdivᵥ_uₕ(-(2 * ᶠρaK_u[colidx] * ᶠstrain_rate[colidx])) /
+            Y.c.ρ[colidx],
+        )
+        @. Yₜ.f.u₃[colidx] -= ᶠinterp_u₃(
+            C3(
+                ᶜdivᵥ(-(2 * ᶠρaK_u[colidx] * ᶠstrain_rate[colidx])) /
+                Y.c.ρ[colidx],
+            ),
+        )
```

Comment:
@szy21 does this make sense? I think it should work.

We may want the same for the `Yₜ.f.u₃` term